enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -835,6 +835,9 @@ func (mc *MigrationCache) Stop() error {
 	return dstShutdownErr
 }
 
+// Compression should only be enabled during a migration if both the source and destination caches support the
+// compressor, in order to reduce complexity around trying to double read/write compressed bytes to one cache and
+// decompressed bytes to another
 func (mc *MigrationCache) SupportsCompressor(ctx context.Context, compressor repb.Compressor_Value, r *resource.ResourceName) bool {
-	return false
+	return mc.src.SupportsCompressor(ctx, compressor, r) && mc.dest.SupportsCompressor(ctx, compressor, r)
```

Comment:
This LGTM at a glance, assuming the underlying cache impls returns a correct value of `SupportsCompressor` during a rollout. There might be other concerns I'm not thinking of though, so it'd be good to get an LGTM from @vadimberezniker and/or @tylerwilliams as well

Reply:
Sounds good

[Distributed cache will return SupportsCompressor=false ](https://github.com/buildbuddy-io/buildbuddy/pull/2962/files#diff-9410f9cf8e952fbcc75b77758e162afd98b0225ec51988fe9e2254a911b34efaR1012)unless all the peers return SupportsCompressor=true. So during a rollout, distributed cache should return SupportsCompressor=false and the migration cache should return SupportsCompressor=false. Once the rollout has proceeded far enough for distributed SupportsCompressor=true, then the migration cache should also return SupportsCompressor=true

Reply:
@vadimberezniker I think this code should be safe for the roll-out with the distributed roll out plan we discussed in that other PR but could you help sanity check me and brandon?